v3/newrelic/tracing.go

```diff
@@ -413,9 +413,7 @@ func (t *txnData) CurrentSpanIdentifier(thread *tracingThread) string {
 
 func (t *txnData) saveSpanEvent(e *spanEvent) {
 	e.AgentAttributes = t.Attrs.filterSpanAttributes(e.AgentAttributes, destSpan)
-	if len(t.SpanEvents) < defaultMaxSpanEvents {
```

Comment:
I am a little surprised there was a hard cap on the span events here, and not an attempt to sample them. That being said, I don't think we should just outright remove this unless we have a very good reason to. It is checking the wrong value though. It should check against the value of run.MaxSpanEvents().

Reply:
As we cannot access the current run in the tracing go file, I set the check against internal.MaxSpanEvents which is the replacement for the deprecated `defaultMaxSpanEvents`